providers/dynadot/api.go

```diff
@@ -0,0 +1,135 @@
+package dynadot
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// API layer for Dynadot
+
+type dynadotProvider struct {
+	key string
+}
+
+type requestParams map[string]string
+
+type header struct {
+	SuccessCode int    `xml:"SuccessCode"`
+	Status      string `xml:"Status"`
+	Error       string `xml:"Error,omitempty"`
+}
+
+type addNsResponse struct {
+	XMLName     xml.Name `xml:"AddNsResponse"`
+	AddNsHeader header   `xml:"AddNsHeader"`
+}
+
+type setNsResponse struct {
+	XMLName     xml.Name `xml:"SetNsResponse"`
+	SetNsHeader header   `xml:"SetNsHeader"`
+}
+
+type getNsResponse struct {
+	XMLName     xml.Name  `xml:"GetNsResponse"`
+	NsContent   nsContent `xml:"NsContent"`
+	GetNsHeader header    `xml:"GetNsHeader"`
+}
+
+type nsContent struct {
+	Host   []string `xml:"Host"`
+	NsName string   `xml:"NsName"`
+}
```

Comment:
I'm not too familiar with go and couldn't find a better way to do this.
For all responses except `get_ns` I only need the header, except it's under a different key each time.

Please let me know if there is a better way to implement this. A current consequence of my implementations means error checking is fully extracted to each method rather than the general `get()`.

Reply:
You did it the way I would have done it!  (So it must be right, eh? :-) )